File: atv4/exerc4.py
class Animal:
    def som(self):
        print("O animal faz um som.")

class Cachorro(Animal):
    def som(self):
        print("O cachorro late.")

class Gato(Animal):
    def som(self):
        print("O gato mia.")

a1 = Cachorro()
a2 = Gato()

a1.som()
a2.som()

File: atv4/exerc4.go
package main
import "fmt"

type Animal interface {
    som()
}

type Cachorro struct{}

func (Cachorro) som() {
    fmt.Println("O cachorro late.")
}

type Gato struct{}

func (Gato) som() {
    fmt.Println("O gato mia.")
}

func main() {
    var a1 Animal = Cachorro{}
    var a2 Animal = Gato{}

    a1.som()
    a2.som()
}
